webapp/scripts/generateHeaders.js

```diff
@@ -16,39 +16,112 @@ const getDomain = function (url) {
+const allowedDomains = [
+  // bitcoin APIs from esplora-client
+  'https://blockstream.info',
+  'https://mempool.space',
+  // EVM rpc urls
+  // Hemi Sepolia
+  'https://int02.testnet.rpc.hemi.network',
+  'https://testnet.rpc.hemi.network',
+  // Hemi Mainnet
+  // Mainnet
+  'https://cloudflare-eth.com',
+  'https://rpc.hemi.network',
+  // Sepolia
+  'https://rpc.sepolia.org',
+  'https://rpc2.sepolia.org',
+  // Needed for rainbow kit
+  'wss://relay.walletconnect.com',
+  'wss://relay.walletconnect.org',
+]
 
-  Header always set X-Frame-Options "SAMEORIGIN"
-  
-  Header always set X-Content-Type-Options "nosniff"
-  
-  Header always set X-XSS-Protection "1; mode=block"
-  
-  Header always set X-Download-Options "noopen"
-
-  Header always set Expect-CT "max-age=86400, enforce"
-
-  Header always set Referrer-Policy "no-referrer-when-downgrade"
-
-  Header always set Permissions-Policy: geolocation=(), microphone=()
-</IfModule>
-`
-
-const domain = getDomain(process.env.NEXT_PUBLIC_ANALYTICS_URL)
+// analytics
+const analyticsDomain = getDomain(process.env.NEXT_PUBLIC_ANALYTICS_URL)
+if (
+  process.env.NEXT_PUBLIC_ENABLE_ANALYTICS === 'true' &&
+  analyticsDomain !== null
+) {
+  allowedDomains.push(`https://${analyticsDomain}`)
+}
+// error-tracking
+const errorTrackingDomain = getDomain(process.env.NEXT_PUBLIC_SENTRY_DSN)
+if (errorTrackingDomain !== null) {
+  allowedDomains.push(`https://${errorTrackingDomain}`)
+}
 
-if (process.env.NEXT_PUBLIC_ENABLE_ANALYTICS === 'true' && domain !== null) {
-  htaccess += `
-<IfModule mod_headers.c>
-    Header always set Access-Control-Allow-Origin "https://${domain}"
-    Header always set Access-Control-Allow-Methods "POST, OPTIONS"
-    Header always set Access-Control-Allow-Headers "Content-Type"
-    Header always set Content-Security-Policy "script-src 'self' "${domain}"
-</IfModule>`
+const serveJson = {
+  headers: [
+    {
+      headers: [
+        {
+          key: 'Strict-Transport-Security',
+          value: 'max-age=31536000; includeSubDomains; preload',
+        },
+        {
+          key: 'X-Frame-Options',
+          value: 'SAMEORIGIN',
+        },
+        {
+          key: 'X-Content-Type-Options',
+          value: 'nosniff',
+        },
+        {
+          key: 'X-XSS-Protection',
+          value: '1; mode=block',
+        },
+        {
+          key: 'X-Download-Options',
+          value: 'noopen',
+        },
+        {
+          key: 'Expect-CT',
+          value: 'max-age=86400, enforce',
+        },
+        {
+          key: 'Referrer-Policy',
+          value: 'no-referrer-when-downgrade',
+        },
+        {
+          key: 'Permissions-Policy',
+          value: 'geolocation=(), microphone=()',
+        },
+        {
+          key: 'Content-Security-Policy',
+          value: `script-src 'self' 'unsafe-inline' ${allowedDomains.join(
+            ' ',
+          )}; connect-src 'self' ${allowedDomains.join(' ')}`,
```

Comment:
I understand this works and is simple. I we would like to improve it, should we split the `allowedDomains` in two lists, one to download scripts from and the other to make requests to?

Reply:
Done [b741e65](https://github.com/hemilabs/ui-monorepo/pull/622/commits/b741e656ecfbc1d701d2abd6785fc7f85d60734f)